### PR TITLE
Feature/ci packing

### PR DESCRIPTION
**Pull request by Kola50011**

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    steps:
+      - name: Install dependencies
+        run: |
+          sudo apt install build-essential cmake git libboost-system-dev libboost-thread-dev file libgtest-dev googletest
+
+      - uses: actions/checkout@v4
+
+      - name: Configure CMake
+        run: cmake -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}} -DCMAKE_CXX_FLAGS="-Werror"
```
Comment (christoph-heiss) [inline .github/workflows/test.yaml]: I'd rather add the `-Werror` directly in the CMake setup add the top to the `add_compile_options(..)` call.

Also, the CI fails, as `cmake` needs to be passed the path to the source directory as first unnamed argument.

```suggestion
      run: cmake -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}} ${{github.workspace}}
```

File: ci/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+cmake -B /libcreate/build -S /libcreate -DCMAKE_BUILD_TYPE=Release -DCMAKE_CXX_FLAGS='-Werror'
```
Comment (christoph-heiss) [inline ci/entrypoint.sh]: Same as above for the CI workflow.

```suggestion
cmake -B /libcreate/build -S /libcreate -DCMAKE_BUILD_TYPE=Release
```

**Review by christoph-heiss — DISMISSED** [review]
Apart from the `-Werror` stuff and the CI failure due to the cmake invocation, LGTM.

File: CMakeLists.txt [hunk 3]
```diff
@@ -1,16 +1,32 @@
...
 #
 
-cmake_minimum_required(VERSION 2.8.12)
-project(libcreate)
+project(
+  libcreate 
+  VERSION ${TAG_VERSION_MAJOR}.${TAG_VERSION_MINOR}.${TAG_VERSION_PATCH}
+)
 
 add_compile_options(-Wall -Wextra -Wpedantic)
```
Comment (christoph-heiss) [inline CMakeLists.txt]:
```suggestion
add_compile_options(-Wall -Wextra -Wpedantic -Werror)
```

**Comment by Kola50011** [comment]
Should be good now. Pls approve and merge

**Review by christoph-heiss — APPROVED** [review]

**Pull request merged**